File: lab/LScrollView/LScrollView/ViewController.m
//
//  ViewController.m
//  LScrollView
//
//  Created by Lanston Peng on 4/26/14.
//  Copyright (c) 2014 Vtm. All rights reserved.
//

#import "ViewController.h"
#import "LScrollView.h"
#import <QuartzCore/QuartzCore.h>
#define DEGREES_TO_RADIANS(angle) ((angle) / 180.0 * M_PI)

@interface ViewController ()<UIScrollViewDelegate>
@property (weak, nonatomic) IBOutlet UIView *indicator;
@property (weak, nonatomic) IBOutlet UIScrollView *scrollView;
@property (weak, nonatomic) IBOutlet UIDatePicker *datePicker;
@end

@implementation ViewController

-(UIColor*)randomColor
{
    int r = arc4random()%3;
    switch (r) {
        case 0:
            return [UIColor purpleColor];
            break;
        case 1:
            return [UIColor greenColor];
            break;
        case 2:
            return [UIColor yellowColor];
            break;
        default:
            break;
    }
    return [UIColor redColor];
}
- (void)viewDidLoad
{
    [super viewDidLoad];
    LScrollView* lScrollView = [[LScrollView alloc]initWithFrame:CGRectMake(0, 0, 320, 300)];
    lScrollView.backgroundColor = [UIColor orangeColor];
    [self.view addSubview:lScrollView];
    
    for(int i = 0 ; i < 2000 ;i++)
    {
        UILabel* l = [[UILabel alloc]initWithFrame:CGRectMake(arc4random()%5 * 50,arc4random()%5* 50, 100, 100)];
        l.text = @"Label";
        l.layer.borderColor = [UIColor blackColor].CGColor;
        l.textAlignment = NSTextAlignmentCenter;
        l.backgroundColor = [self randomColor];
        [lScrollView addSubview:l];
    }
    

    
    
    double rads = DEGREES_TO_RADIANS(45);
    //lScrollView.layer.transform = CATransform3DMakeRotation(rads, 0, 0, 1);
    /*
    UILabel* l2 = [[UILabel alloc]initWithFrame:CGRectMake(100, 160, 100, 100)];
    l2.text = @"Label";
    l2.layer.borderColor = [UIColor blackColor].CGColor;
    l2.textAlignment = NSTextAlignmentCenter;
    l2.backgroundColor = [UIColor purpleColor];
    [lScrollView addSubview:l2];
     */
    /*
    //self.datePicker.frame = CGRectMake(0, -self.datePicker.frame.size.height, self.datePicker.frame.size.width, self.datePicker.frame.size.height);
    self.scrollView.backgroundColor = [UIColor colorWithRed:0 green:0 blue:0 alpha:0.4f];
    self.scrollView.contentSize = CGSizeMake(320, self.scrollView.frame.size.height + 200);
    self.scrollView.bounces = YES;
    self.indicator.backgroundColor = [UIColor orangeColor];
    [self.indicator removeConstraints:self.indicator.constraints];
    self.indicator.frame = CGRectMake(0, 0, 320, 600);
     */
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: lab/PullToRefresh/PullToRefresh/ViewController.swift
//
//  ViewController.swift
//  PullToRefresh
//
//  Created by Lanston Peng on 9/15/14.
//  Copyright (c) 2014 Vtm. All rights reserved.
//

import UIKit

class ViewController: UIViewController,UITableViewDataSource,UITableViewDelegate {

    @IBOutlet weak var tableView: UITableView!
    var refreshView:RefreshView?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.refreshView = RefreshView(frame: CGRectMake(0, -120, self.view.bounds.width, 120))
        self.tableView.backgroundColor = UIColor.purpleColor()
        //self.tableView.alpha = 0.4
        self.view.insertSubview(refreshView!, aboveSubview: self.tableView)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    func scrollViewDidScroll(scrollView: UIScrollView) {
        refreshView?.ScrollViewDidScroll(scrollView)
    }
    
    //MARK:-
    //MARK:Datasource
    func numberOfSectionsInTableView(tableView: UITableView) -> Int {
        return 1
    }
    
    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return 20
    }
    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCellWithIdentifier("customCell") as UITableViewCell
        return cell
    }
    
    func tableView(tableView: UITableView, heightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat {
        return 80
    }
}

